ldap_jwt_auth/routers/login.py

```diff
@@ -0,0 +1,52 @@
+"""
+Module for providing an API router which defines login route(s).
+"""
+import logging
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.responses import JSONResponse
+
+from ldap_jwt_auth.auth.authentication import Authentication
+from ldap_jwt_auth.auth.jwt_hanlder import JWTHandler
+from ldap_jwt_auth.core.config import config
+from ldap_jwt_auth.core.exceptions import InvalidCredentialsError, LDAPServerError
+from ldap_jwt_auth.core.models import UserCredentials
+
+logger = logging.getLogger()
+
+router = APIRouter(prefix="/login", tags=["authentication"])
+
+
+@router.post(
+    path="/",
+    summary="Login with a username and password",
+    response_description="A JWT access token including a refresh token as HTTP only cookie",
+)
+def login(
+    user_credentials: UserCredentials, authentication: Authentication = Depends(), jwt_handler: JWTHandler = Depends()
```

Comment:
Could these be annotations?

Reply:
Addressed in https://github.com/ral-facilities/ldap-jwt-auth/pull/7/commits/6f6e8f989435e6ca8dec3e0910274eca13b75d8b